### PR TITLE
Radar controller/accumulator

### PR DESCRIPTION
**Pull request by frk2**
Theres a new launch file now. You can launch this with:

`launch src/launch/all_launch.launch`

you can observe radar topics with:

`ros2 topic echo /radar_tracks`

### CODE REVIEW

File: src/radar/radar/radar_controller.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    (0x161, ECU.DSU, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA), 1, 7,
+     '\x00\x1e\x00\x00\x00\x80\x07'),
+    (0x365, ECU.DSU, (CAR.RAV4, CAR.COROLLA), 0, 20, '\x00\x00\x00\x80\xfc\x00\x08'),
+    (0x366, ECU.DSU, (CAR.RAV4, CAR.COROLLA), 0, 20, '\x00\x72\x07\xff\x09\xfe\x00'),
+    (0x4CB, ECU.DSU, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA), 0, 100,
+     '\x0c\x00\x00\x00\x00\x00\x00\x00'),
+]
+
+
+class RadarController(Node):
```
Comment (J-Rojas) [inline src/radar/radar/radar_controller.py]: Would be good to change this class name to something like ToyotaRadarController since it only works with Toyota radar units.

File: src/opendbc/setup.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+setup(
+    name='opendbc',
+    version='0.0.1',
+    packages=find_packages(exclude=['test']),
+    package_data={
+        'opendbc': ['*.dbc']
+    },
+    install_requires=['setuptools'],
+    # data_files=[
```
Comment (J-Rojas) [inline src/opendbc/setup.py]: Any value to keeping this commented out?

File: src/radar/radar/radar_controller.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                if 528 <= can_msg.id <= 543:
+                    track = RadarTrack(track_id=can_msg.id - 528,
+                                       counter=msg["COUNTER"],
+                                       lat_dist=msg["LAT_DIST"],
+                                       lng_dist=msg["LONG_DIST"],
+                                       new_track=bool(msg["NEW_TRACK"]),
+                                       valid=bool(msg["VALID"]))
+                    self.radar_tracks_msg.radar_tracks.append(track)
+                elif 544 <= can_msg.id <= 559:
+                    accel = RadarTrackAccel(track_id=can_msg.id - 544,
```
Comment (J-Rojas) [inline src/radar/radar/radar_controller.py]: Where does speed get decoded?

File: src/radar/radar/radar_controller.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+class RadarController(Node):
+    """
+    This radar controller is hardcoded to work only with the Toyota Corolla/Rav4/Camry 2017 Denso unit
+    """
+
+    def __init__(self):
+        super().__init__('radar')
+        # self.tracks_pub = self.create_publisher(String, 'radar_tracks')
+        self.can_pub = self.create_publisher(CanMessage, 'can_send')
+        self.adas_db = cantools.db.load_file(os.path.join(opendbc.DBC_PATH, 'toyota_prius_2017_adas.dbc'))
```
Comment (J-Rojas) [inline src/radar/radar/radar_controller.py]: Where are these dbc files located?

Comment (frk2) [reply src/radar/radar/radar_controller.py]: in the opendbc submodule. Make sure you did a git submodule update --init

File: src/radar/radar/toyota_radar_controller.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    def __init__(self):
+        super().__init__('radar')
+        # self.tracks_pub = self.create_publisher(String, 'radar_tracks')
+        self.can_pub = self.create_publisher(CanMessage, 'can_send')
+        self.adas_db = cantools.db.load_file(os.path.join(opendbc.DBC_PATH, 'toyota_prius_2017_adas.dbc'))
+        self.db = cantools.db.load_file(os.path.join(opendbc.DBC_PATH, 'toyota_prius_2017_pt_generated.dbc'))
+
+        self.can_sub = self.create_subscription(CanMessage, 'can_recv', self.on_can_message)
+        self.radar_pub = self.create_publisher(RadarTracks, 'radar_tracks')
+        self.power_on_timer = self.create_timer(1.0/100, self.power_on_radar)
```
Comment (J-Rojas) [inline src/radar/radar/toyota_radar_controller.py]: Is this a persistent timer constantly executing power_on_radar()? If so, why is this necessary?

File: src/radar/radar/toyota_radar_controller.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                    vl += chr(cnt)
+                tosend = bytearray()
+                tosend.extend(map(ord, vl))
+                message = CanMessage(id=addr, interface=CanMessage.CANTYPE_RADAR, data=tosend)
+                self.can_pub.publish(message)
+        self.frame += 1.
+
+    def on_can_message(self, can_msg):
+        if can_msg.interface == CanMessage.CANTYPE_RADAR:
+            if 528 <= can_msg.id <= 559:
```
Comment (J-Rojas) [inline src/radar/radar/toyota_radar_controller.py]: Would be best to move these indexes to class constants. They are duplicated in several places.

File: src/radar/radar/toyota_radar_controller.py (hunk 5 above)
Comment (frk2) [reply src/radar/radar/toyota_radar_controller.py]: Toyota wants it this way :man_shrugging:

File: src/radar/radar/toyota_radar_controller.py (hunk 6 above)
Comment (frk2) [reply src/radar/radar/toyota_radar_controller.py]: I'll put a FIXME in there for now to take care of the aesthetic part later once stuff works. The radar IDs are only used here.

Comment (frk2) [reply src/radar/radar/toyota_radar_controller.py]: Actually i just tried doing this and honestly it becomes more confusing. The stuff thats coming off the wire is gonna be the ID. I would rather have the ID just hardcoded right there so it avoids me having to do a mental lookup of a constant.

File: src/radar/radar/toyota_radar_controller.py (hunk 5 above)
Comment (J-Rojas) [reply src/radar/radar/toyota_radar_controller.py]: Could you add a TODO comment here about how it's not clear why this is needed but that it does continually fire? Definitely should look into why this is at some point... it's sending out 100 messages a second just to keep the radar 'on'.

BTW Timers could be one-shot, but the default mode is perpetual.
http://wiki.ros.org/roscpp/Overview/Timers

Comment (frk2) [reply src/radar/radar/toyota_radar_controller.py]: Feel free to add one and lets merge this PR - need to actually test it.
